sdk/spring/spring-cloud-azure-actuator/src/main/java/com/azure/spring/cloud/actuator/cosmos/CosmosHealthIndicator.java

```diff
@@ -44,19 +45,33 @@ public CosmosHealthIndicator(CosmosAsyncClient cosmosAsyncClient, String databas
     }
 
     @Override
-    protected void doHealthCheck(Builder builder) {
-        CosmosDatabaseResponse response = this.cosmosAsyncClient.getDatabase(database)
-            .read()
-            .block(timeout);
-
-        if (response != null) {
-            LOGGER.info("The health indicator cost {} RUs, cosmos uri: {}, dbName: {}",
-                response.getRequestCharge(), endpoint, database);
-        }
-        if (response == null) {
-            builder.down();
+    protected void doHealthCheck(Builder builder) throws Exception {
+        if (database == null) {
+            builder.status("The option of `spring.cloud.azure.cosmos.database` is not configured!");
         } else {
-            builder.up().withDetail("database", response.getProperties().getId());
+            try {
+                CosmosDatabaseResponse response = this.cosmosAsyncClient.getDatabase(database)
+                                                                        .read()
+                                                                        .block(timeout);
+
+                if (response != null) {
+                    LOGGER.info("The health indicator cost {} RUs, cosmos uri: {}, dbName: {}",
+                        response.getRequestCharge(), endpoint, database);
+                    builder.withDetail("RUs", response.getRequestCharge());
+                    builder.withDetail("CosmosUri", endpoint);
+                    builder.up().withDetail("database", response.getProperties().getId());
```

Comment:
How about letting `LOGGER.info(...)` and `builder.up().withDetail()` output the same information. Use `database` instead of `response.getProperties().getId()`.

Reply:
yes